src/main.rs

```diff
@@ -367,15 +361,43 @@ fn main() {
     }
     #[cfg(windows)]
     {
+        use future::Either;
+        use futures::{future::IntoFuture, Async};
+
+        // Here to satisfy 'static/Send constraints imposed by block_on method.
+        let config_watcher = std::sync::Arc::new(std::sync::Mutex::new(config_watcher));
+
         let ctrl_c = tokio_signal::ctrl_c().flatten_stream().into_future();
         let crash = future::poll_fn(move || graceful_crash.poll());
 
-        let interruption = rt
-            .block_on(ctrl_c.select2(crash))
-            .map_err(|_| ())
-            .expect("Neither stream errors");
+        let mut ctrl_c_select = ctrl_c.select2(crash);
+        let interruption = loop {
+            let config_watcher = config_watcher.clone();
+            let change = future::poll_fn(move || {
```

Comment:
this feels like this code could be moved into the watcher.rs file? I think that might clean this up a bit. I'd be nice to see if we can reduce some of this duplication.

Reply:
True